### ISSUE TITLE
Inconsistency in virtual model design: $parent property

### ISSUE THREAD

**Issue by b1r3k** [ca77y/angular-ya-treeview#5]
Default structure of virtual node looks somewhat like that:
$$hashKey: "00K"
$children: Array[3]
$hasChildren: true
$model: Object
$parent: Object

If one would check how $children property looks like, one would get something like this:
$children: Array[3]
0: Object
$$hashKey: "03E"
$children: Array[5]
$hasChildren: true
$model: Object
$parent: Object
collapsed: true
**proto**: Object
1: Object
$$hashKey: "03F"
$children: Array[3]
$hasChildren: true
$model: Object
$parent: Object
collapsed: true
**proto**: Object

In other words, $children links to other virtual nodes. $parent however links to real node, not the virtual one. Such design is non-intuitive and particularly frustrating while trying to extend behaviour of tree by manipulating virtual tree model directly.
